### ISSUE TITLE
Make a proof-of-concept of whitelabel engine

### ISSUE THREAD

**Issue by Mehonoshin** [Mehonoshin/smartvpn-billing#111]
We need to think about adding a set of entities for whitelabel partner program.

**Comment by MrConstantine** [comment]
You need suggestions? Are you going to add this to the billing functionality?

**Comment by Mehonoshin** [comment]
@MrConstantine yep. First of all I want to hear if it is really needed. Of course it will be an optional feature.

**Comment by MrConstantine** [comment]
It would be great. I think that such functionality could promote loyalty to customers. For example, you can make a promotion in the form of discounts (1 registered customer = 2% - 5% discount)

**Comment by MrConstantine** [comment]
Or it can be a reward in the form of bonus points with which can then pay for VPN. For example, 1 client by referral link = 15 points (15 points = 15 rubles or $ 0.22)

**Comment by Mehonoshin** [comment]
@MrConstantine we already have referral link implemented, but it does not give discounts, it allows to collect money on the account.
This specific issue is about whitelabel program, when user can add his own DNS, logo and have an illusion of own VPN service.
